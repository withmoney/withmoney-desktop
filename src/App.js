import React from 'react';
import Login from './pages/Login';

const App = () => (
  <Login />
);

export default App;
