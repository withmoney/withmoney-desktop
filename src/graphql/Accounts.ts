import { gql } from '@apollo/client';

// query
export const GET_ACCOUNTS = gql`
  query getAccounts {
    me {
      accounts(where: { deletedAt: { equals: null } }) {
        id
        name
        currency
      }
    }
  }
`;

// mutations
export const CREATE_ACCOUNT = gql`
  mutation createAccount($name: String!, $currency: Currency!) {
    createOneAccount(data: { name: $name, currency: $currency }) {
      id
      name
    }
  }
`;

export const UPDATE_ACCOUNT = gql`
  mutation updateAccount($id: String!, $name: String!, $currency: Currency!) {
    updateOneAccount(where: { id: $id }, data: { name: $name, currency: $currency }) {
      id
      name
    }
  }
`;

export const DELETE_ACCOUNT = gql`
  mutation deleteAccount($id: String!) {
    deleteOneAccount(where: { id: $id }) {
      id
      name
    }
  }
`;

export const RESTORE_ACCOUNT = gql`
  mutation restoreAccount($id: String!) {
    restoreOneAccount(where: { id: $id }) {
      id
      name
    }
  }
`;
