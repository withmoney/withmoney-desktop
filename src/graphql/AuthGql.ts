import { gql } from '@apollo/client';

//Query
export const GET_ME = gql`
  query getMe {
    me {
      id
    }
  }
`;

export const GET_OPERATIONS = gql`
  query getOperations($startDateTime: DateTime!, $endDateTime: DateTime!, $accountId: String!) {
    me {
      operations(
        where: {
          createdAt: { gte: $startDateTime, lte: $endDateTime }
          accountId: { equals: $accountId }
        }
      ) {
        id
        name
        value
        type
        isPaid
        createdAt
      }
    }
  }
`;

export const GET_ACCOUNTS = gql`
  query getAccounts {
    me {
      accounts {
        id
        name
      }
    }
  }
`;

//Mutation
export const USER_LOGIN = gql`
  mutation userLogin($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      token
    }
  }
`;

export const USER_REGISTER = gql`
  mutation userRegister(
    $firstName: String!
    $lastName: String!
    $email: String!
    $password: String!
  ) {
    register(
      user: { firstName: $firstName, lastName: $lastName, email: $email, password: $password }
    )
  }
`;

export const CHECK_HASH = gql`
  mutation checkHash($hash: String!) {
    checkHashEmail(hash: $hash)
  }
`;

export const REQUEST_CHANGE_PASSWORD = gql`
  mutation requestChangePassword($email: String!) {
    requestChangePassword(email: $email)
  }
`;

export const CHANGE_PASSWORD = gql`
  mutation changePassword($hash: String!, $password: String!) {
    changePassword(hash: $hash, password: $password)
  }
`;
