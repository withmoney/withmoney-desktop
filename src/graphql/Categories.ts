import { gql } from '@apollo/client';

// query
export const CATEGORY_SEARCH = gql`
  query categorySearch($name: String!, $type: TransactionType!) {
    categories: findManyCategory(where: { name: { contains: $name }, type: { equals: $type } }) {
      id
      name
      type
    }
  }
`;

export const ALL_CATEGORY = gql`
  query getAllCategories {
    categories: findManyCategory(orderBy: [{ name: asc }]) {
      id
      name
      type
    }
  }
`;

// mutations
export const CREATE_CATEGORY = gql`
  mutation createCategory($name: String!, $type: TransactionType!) {
    createOneCategory(data: { name: $name, type: $type }) {
      id
      name
    }
  }
`;
