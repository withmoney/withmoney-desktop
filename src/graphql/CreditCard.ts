import { gql } from '@apollo/client';

// Queries
export const CREDIT_CARDS = gql`
  query getAllCards($id: String!) {
    allCreditCards: findManyCreditCard(
      where: { accountId: { equals: $id }, deletedAt: { equals: null } }
      orderBy: { name: asc }
    ) {
      data {
        id
        name
        brand
        limit
      }
    }
  }
`;

export const FILTER_CREDIT_CARD = gql`
  query filterCard($name: String!, $id: String!) {
    creditCards: findManyCreditCard(
      where: { accountId: { equals: $id }, name: { contains: $name }, deletedAt: { equals: null } }
    ) {
      data {
        id
        name
        brand
        limit
      }
    }
  }
`;

// Mutations

export const CREATE_CREDIT_CARD = gql`
  mutation createCreditCard(
    $name: String!
    $limit: Float!
    $brand: CreditCardBrand!
    $account: String!
  ) {
    createOneCreditCard(data: { name: $name, limit: $limit, brand: $brand, accountId: $account }) {
      id
      name
      limit
      brand
    }
  }
`;
