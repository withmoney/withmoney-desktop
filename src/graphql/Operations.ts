import { gql } from '@apollo/client';

// query
export const GET_OPERATIONS = gql`
  query getOperations($startDateTime: DateTime!, $endDateTime: DateTime!, $accountId: String!) {
    me {
      operations(
        where: {
          paidAt: { gte: $startDateTime, lte: $endDateTime }
          accountId: { equals: $accountId }
        }
        orderBy: [{ paidAt: asc }]
      ) {
        id
        name
        value
        type
        isPaid
        paidAt
        createdAt
        account {
          id
          name
        }
        category {
          id
          name
        }
      }
    }
  }
`;
// mutations

export const UPDATE_OPERATION = gql`
  mutation updateOperation(
    $id: String!
    $name: String!
    $categoryId: String!
    $accountId: String!
    $type: TransactionType!
    $value: Float!
    $isPaid: Boolean!
    $paidAt: DateTime!
  ) {
    updateOneOperation(
      where: { id: $id }
      data: {
        name: $name
        type: $type
        accountId: $accountId
        categoryId: $categoryId
        value: $value
        isPaid: $isPaid
        paidAt: $paidAt
      }
    ) {
      id
      name
    }
  }
`;
