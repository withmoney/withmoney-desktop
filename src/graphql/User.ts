import { gql } from '@apollo/client';

export const UPDATE_USER = gql`
  mutation updateUser($firstName: String!, $lastName: String!, $email: String!, $language: Locale) {
    updateOneUser(
      data: { firstName: $firstName, lastName: $lastName, email: $email, language: $language }
    ) {
      id
      firstName
      lastName
      email
      language
    }
  }
`;
