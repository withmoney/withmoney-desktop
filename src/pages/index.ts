export { default as Dashboard } from './Dashboard';
export { default as Login } from './Login';
export { default as ChangePassword } from './ChangePassword';
export { default as RegistrationVerify } from './RegistrationVerify';
export { default as ResetPassword } from './ResetPassword';
export { default as SignUp } from './SignUp';
export { default as Reports } from './Reports';
