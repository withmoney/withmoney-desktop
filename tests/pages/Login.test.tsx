import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import { USER_LOGIN } from 'src/graphql/AuthGql';
import { Login } from 'src/pages';

const mocks = [
  {
    request: {
      query: USER_LOGIN,
      variables: {
        email: 'davidcostadev@gmail.com',
        password: 'password',
      },
    },
    result: {
      data: {
        login: {
          token: 'asdf',
        },
      },
    },
  },
];

describe('Login page', () => {
  it('Should return login form correctly', () => {
    const { container } = render(
      <MockedProvider mocks={mocks}>
        <Login />
      </MockedProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
