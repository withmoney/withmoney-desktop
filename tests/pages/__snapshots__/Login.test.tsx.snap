// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page Should return login form correctly 1`] = `
.c0 {
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #f2f2f2;
}

.c1 {
  padding-left: 15px;
  padding-right: 15px;
}

.c2 {
  text-align: center;
  font-size: 42px;
  font-weight: 700;
  margin-top: 70px;
  margin-bottom: 45px;
}

.c3 {
  background-color: white;
  margin-left: auto;
  margin-right: auto;
  max-width: 445px;
  width: 100%;
  border-radius: 5px;
  padding: 20px 20px 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  font-size: 32px;
  font-weight: 700;
  text-align: center;
  margin-bottom: 20px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  font-size: 16px;
  border-radius: 5px;
  border: 1px solid #bdbdbd;
  padding: 10px 12px;
  margin-bottom: 20px;
  outline: none;
}

.c5:focus {
  box-shadow: 0 0 0px 2px rgba(33,150,83,0.5);
}

.c5:disabled {
  color: gray;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 + .c6 {
  margin-top: 20px;
}

.c9 {
  background-color: #219653;
  color: white;
  font-size: 16px;
  border-width: 0px;
  padding: 11px 13px;
  border-radius: 5px;
  cursor: pointer;
  outline: none;
}

.c9:hover {
  background-color: #308856;
}

.c9:focus {
  box-shadow: 0 0 0px 2px rgba(33,150,83,0.5);
}

.c9:active {
  background-color: #256a43;
}

.c9:disabled {
  background-color: hsla(146,10%,36%,1);
}

.c8 {
  color: #219653;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c8:hover {
  color: #308856;
}

.c8:active {
  color: #256a43;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        withmoney
      </h1>
      <form
        class="c3"
      >
        <h2
          class="c4"
        >
          Signin
        </h2>
        <input
          class="c5"
          name="email"
          placeholder="Email"
          required=""
          type="email"
        />
        <input
          class="c5"
          name="password"
          placeholder="Password"
          required=""
          type="password"
        />
        <div
          class="c6 c7"
        >
          <a
            class="c8"
          >
            Reset your password
          </a>
          <button
            class="c9"
          >
            Login
          </button>
        </div>
        <div
          class="c6 c7"
        >
          <span>
            Do you not have an account?
          </span>
          <a
            class="c8"
          >
            Signup
          </a>
        </div>
      </form>
    </div>
  </div>
</div>
`;
